### PR TITLE
Poseidon2 Custom Matrix (not MDS)

### PR DESCRIPTION
**Pull request by npwardberkeley**
To address #134

### CODE REVIEW

**Comment by dlubarov** [comment]
I'm traveling soon, could someone else take the review? cc @jtguibas @vivekvpandya @jorgeantonio21

**Review by vivekvpandya — APPROVED** [review]
Looks good. However there is no test which should have detect this earlier? i.e trying yo compare result with other implementation ?

**Comment by npwardberkeley** [comment]
Good idea! Added a test, currently debugging why it fails

**Review by jorgeantonio21 — COMMENTED** [review]
The PR looks like it is going in the right direction, the matrix permutation method looks correct to me. Any other bugs found so far ? @npwardberkeley @dlubarov

File: poseidon2/Cargo.toml [hunk 1]
```diff
@@ -5,13 +5,15 @@ edition = "2021"
 license = "MIT OR Apache-2.0"
 
 [dependencies]
+ark-ff = { version = "^0.4.0", default-features = false }
```
Comment (dlubarov) [inline poseidon2/Cargo.toml:8]: Looks like these two deps can be in dev-dependencies below

**Review by dlubarov — APPROVED** [review]
LGTM

**Pull request merged**